megatron/training.py

```diff
@@ -297,6 +350,9 @@ def get_optimizer(model, neox_args):
         _param_groups.append(param_group)
     param_groups = _param_groups
 
+    # If we're using mup, then the optimizer must be adam or sgd
+    assert not neox_args.use_mup or (neox_args.optimizer_type.lower() == "adam" or neox_args.optimizer_type.lower() == "sgd")
```

Comment:
Let's print a message if this assertion fails so that they don't have to read this comment. 

assert not neox_args.use_mup or (neox_args.optimizer_type.lower() == "adam" or neox_args.optimizer_type.lower() == "sgd"), f'If use_mup == True, you must specify either the adam or sgd optimizers. You passed: {neox_args.optimizer_type.lower()}'